firestore/options.go

```diff
@@ -170,3 +170,52 @@ func processSetOptions(opts []SetOption) (fps []FieldPath, all bool, err error)
 		return nil, false, fmt.Errorf("conflicting options: %+v", opts)
 	}
 }
+
+type runQuerySettings struct {
+	// Explain options for the query. If set, additional query
+	// statistics will be returned. If not, only query results will be returned.
+	explainOptions *pb.ExplainOptions
+}
+
+// newRunQuerySettings creates a runQuerySettings with a given RunOption slice.
+func newRunQuerySettings(opts []RunOption) (*runQuerySettings, error) {
+	s := &runQuerySettings{}
+	for _, o := range opts {
+		if o == nil {
+			return nil, errors.New("firestore: RunOption cannot be nil")
+		}
+		err := o.apply(s)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return s, nil
+}
+
+// RunOption are options used while running a query
+type RunOption interface {
+	apply(*runQuerySettings) error
+}
+
+// ExplainOptions is options used to configure explain query.
+//
+// Query Explain feature is still in preview and not yet publicly available.
+// Pre-GA features might have limited support and can change at any time.
+type ExplainOptions struct {
+	// When false (the default), the query will be planned but not executed, returning only
```

Comment:
We can have that but I would prefer not to deviate from the proto's terminology

Reply:
I don't care about the proto, since users won't see those, but the [doc](https://cloud.google.com/firestore/docs/query-explain) does use that terminology, so I reluctantly agree with you.